tests/test_callprofiler.py

```diff
@@ -380,4 +380,47 @@ def foo():
         modmesh.call_profiler.reset()
         foo()
 
+    def test_status(self):
+        modmesh.wrapper_profiler_status.disable()
+        self.assertFalse(modmesh.wrapper_profiler_status.enabled)
+
+        modmesh.call_profiler.reset()
+        buf = modmesh.ConcreteBuffer(10)
+        result = modmesh.call_profiler.result()
+        self.assertEqual({}, result)
+
+        modmesh.wrapper_profiler_status.enable()
+        modmesh.call_profiler.reset()
+        buf = modmesh.ConcreteBuffer(10)  # noqa: F841
+        result = modmesh.call_profiler.result().get("children")
+        names = [data["name"] for data in result]
+        self.assertEqual(['ConcreteBuffer.__init__'], names)
+
+    def test_names(self):
+        modmesh.call_profiler.reset()
+        buf = modmesh.ConcreteBuffer(10)
+        buf2 = buf.clone()  # noqa: F841
+        result = modmesh.call_profiler.result().get("children")
+        names = [data["name"] for data in result]
+        expected_name = ['ConcreteBuffer.__init__', 'ConcreteBuffer.clone']
+        self.assertEqual(expected_name, names)
+
+    def test_entry(self):
+        modmesh.call_profiler.reset()
+        buf = modmesh.ConcreteBuffer(10)
+        buf2 = buf.clone()  # noqa: F841
+        buf2 = buf.clone()  # noqa: F841
+        result = modmesh.call_profiler.result().get("children")
+        init_result = list(filter(
+            lambda d: d["name"] == "ConcreteBuffer.__init__", result
+        ))[0]
+        clone_result = list(filter(
+            lambda d: d["name"] == "ConcreteBuffer.clone", result)
+        )[0]
+
+        self.assertEqual(1, init_result["count"])
+        self.assertGreater(init_result["total_time"], 0)
+        self.assertEqual(2, clone_result["count"])
+        self.assertGreater(clone_result["total_time"], 0)
```

Comment:
In TimeRegistry, it have a function named `entry` to fetch specific result with specific function name, but CallProfiler doesn't. Therefore, I use filter to fetch specific result.

```diff
-       modmesh.time_registry.entry('ConcreteBuffer.__init__')
+       result = modmesh.call_profiler.result().get("children")
+       init_result = list(filter(
+           lambda d: d["name"] == "ConcreteBuffer.__init__", result
+       ))[0]
```

Same, TimeRegistry can specific attribute to fetch value, but CallProfiler doesn't. Therefore, I use key to fetch value.

```diff
-       modmesh.time_registry.entry('ConcreteBuffer.__init__').count
+       init_result["count"]
```